### PR TITLE
change mongo version to 2.6.4

### PR DESCRIPTION
**Pull request by clytwynec**
@jzoldak  @feanil  Please review.  I've tested this on devstack.  The mongo role is working, but I'm getting some weird, seemingly unrelated errors afterward.

Once this is merged,  I can update the jenkins worker AMI.

### CODE REVIEW

File: playbooks/roles/mongo/tasks/main.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 - name: install mongo server and recommends
   apt: >
-    pkg=mongodb-10gen={{ mongo_version }}
+    pkg=mongodb-org={{ mongo_version }}
```
Comment (clytwynec) [inline playbooks/roles/mongo/tasks/main.yml:21]: For reasoning behind this: http://docs.mongodb.org/manual/tutorial/install-mongodb-on-ubuntu/#considerations

"""Changed in version 2.6: The package structure and names have changed as of version 2.6. For instructions on installation of an older release, please refer to the documentation for the appropriate version."""

**Comment by jzoldak** [comment]
@clytwynec you're getting a weird error where?

**Comment by clytwynec** [comment]
@jzoldak  I attempted to provision devstack twice: 
- The first time I got this error: https://gist.github.com/clytwynec/a455dc7d024188b84467
- The second time I got this one: https://gist.github.com/clytwynec/58cdd0272167b2919a46

Any ideas?

**Comment by jzoldak** [comment]
Code looks good to me. :+1: 
Since you cannot successfully provision devstack successfully prior to this change, I don't think that should be a gating factor for merging this in.

**Comment by clytwynec** [comment]
@feanil  Is this ok to merge?

**Comment by feanil** [comment]
@clytwynec  Please add an entry into the change log since this will be a breaking change for anyone who is currently installing mongo using the mongo role.

**Comment by carsongee** [comment]
Any chance we could mod the play to handle upgrading it from 2.4?  It looks like it is just a drop in replacement upgrade procedure: http://docs.mongodb.org/manual/release-notes/2.6-upgrade/

That can be out of scope for this if it needs to be, but I'd at least be curious as to what happens if you run this on a devstack that already has 2.4

**Comment by clytwynec** [comment]
@carsongee  I think we could do that, I was just taking the quickest route to being able to get a new AMI for our jenkins workers that has 2.6.4.  I could build the AMI off of this branch, and wait to merge this?

@jzoldak Thoughts? ^

**Comment by jzoldak** [comment]
@clytwynec I'd go with building the jenkins worker AMI off this branch and wait to merge this until after you can test what happens with a devstack 2.4 -> 2.6.

**Pull request closed without merging**